### PR TITLE
feat: add universal endpoint doc

### PR DESCRIPTION
**Pull request by annielz**
### Description

Add doc for universal endpoint 

### Rationale

To meet the requirement specified in Greenfield Whitepaper here: https://github.com/bnb-chain/greenfield-whitepaper/blob/main/part3.md#2312-https-rest-api

### CODE REVIEW

File: src/guide/storage-provider/services/gateway.md [hunk 1]
```diff
@@ -30,5 +30,18 @@ service quality.
 ### Micro Service Clients
 Various microservice clients inside SP, including Uploader, Downloader, etc.
 
+### Universal Endpoint
+Universal endpoint provides url for download public files without authentication needed. It supports using any
```
[the remark this answers is not in the document]
Comment (annielz) [reply src/guide/storage-provider/services/gateway.md]: updated

File: src/guide/storage-provider/services/gateway.md [hunk 2]
```diff
@@ -30,5 +30,22 @@ service quality.
 ### Micro Service Clients
 Various microservice clients inside SP, including Uploader, Downloader, etc.
 
+### Universal Endpoint
+Universal endpoint supports using any valid endpoint in the download url, and will automatically redirect to the correct endpoint contains
+the object for download.
+
+For instance, when user access an example endpoint greenfield.sp1.com of SP1 in some environment, the request url will be:
```
Comment (clydemeng) [inline src/guide/storage-provider/services/gateway.md:37]: 1.  users access
2. Can you revise the term "greenfield.sp1.com of SP1 in some environment" ?
3. Can you include more details from white paper and more tech details?

File: src/guide/storage-provider/services/gateway.md [hunk 3]
```diff
@@ -30,5 +30,58 @@ service quality.
...
+#### View File
+If you want to view a file using Universal Endpoint without download, the url is like: https://greenfield.sp1.com/view/mybucket/myobject.jpg.
+This is enforced by adding to the header:
+```md
+    Content-Disposition=inline
+```
+
+#### Public File Access
+Public files can be downloaded/viewed with the following points to notice:
+1. Downloader/viewer's quota will not be deducted, but the file provider's quota will be deducted per download/view.
```
Comment (Fynnss) [inline src/guide/storage-provider/services/gateway.md:80]: `file provider` => `object owner`

File: src/guide/storage-provider/services/gateway.md (hunk 2 above)
Comment (annielz) [reply src/guide/storage-provider/services/gateway.md]: updated

File: src/guide/storage-provider/services/gateway.md (hunk 3 above)
Comment (annielz) [reply src/guide/storage-provider/services/gateway.md]: fixed

**Review by ruojunm — APPROVED** [review]

**Pull request merged**